react-native-google-cast/react-native-google-cast
# Fix linker warnings from iOS react-native-google-cast

### Background
We are getting a lot of linker warnings from iOS react-native-google-cast.
<img src='https://user-images.githubusercontent.com/41613812/107552983-de362680-6bdc-11eb-9cf2-da502955cf6b.png' width=400px  />
I've tried to search the reason for these warnings and seems like the cause is not creating a proper header file for each category implementation in **`types`** folder.

### How to test the issue?
1. Open **`playground`** folder. Run **`yarn install`**.
2. Go to **`ios`** and run **`pod install`**.
3. Open the Xcode project.
4. Build the project. 
5. See the warnings.

### What has been done?
1. I've created a header file for each implementation file for **`types`** folder.
2. I've imported header files instead of **`.m`** files.

### How to test the changes?
1. Open **`playground`** folder. Run **`yarn install`**.
2. Go to **`ios`** and run **`pod install`**.
3. Open the Xcode project.
4. Build the project. 
5. Check that no more warnings are coming from **`react-native-google-cast`** or at least from **`types`** folder files 😅.
6. Check that I've used the right methods declarations in the header files.
